### PR TITLE
jameica: add `arch64-darwin` support

### PR DESCRIPTION
**Pull request by fbettag**
tested on macos x86_64 and aarch64.

### CODE REVIEW

**Comment by fbettag** [comment]
now bundles the .app nicely an only on mac.

tested on Seqoia aarch64 (m1)

File: pkgs/applications/office/jameica/default.nix [hunk 1]
```diff
@@ -76,6 +77,13 @@ stdenv.mkDerivation rec {
     install -Dm644 build/jameica-icon.png $out/share/pixmaps/jameica.png
     cp ${desktopItem}/share/applications/* $out/share/applications/
 
+    # Create .app bundle for macOS
```
Comment (flokli) [inline pkgs/applications/office/jameica/default.nix:80]: Can you end the multiline string after the cp ${desktopItem} line, and then concatenate a lib.optionalString with the darwin-specific bits? This seems more consistent with the rest of nixpkgs, especially when conditionally adding other multiline strings.

Comment (fbettag) [reply pkgs/applications/office/jameica/default.nix:81]: thanks!

**Comment by flokli** [comment]
Squashed things together, and rebased on top of master.

**Comment by fbettag** [comment]
thanks man!

**Pull request merged**